core/src/com/unciv/logic/civilization/QuestManager.kt

```diff
@@ -79,13 +79,42 @@ class QuestManager {
 
         decrementQuestCountdowns()
 
-        handleGlobalQuests()
+        handleExpiredGlobalQuests()
         handleIndividualQuests()
 
         tryStartNewGlobalQuest()
         tryStartNewIndividualQuests()
     }
 
+    /** Get notified that [killer] cleared a Barbarian Encampement on [tile] */
+    fun campKillNotification(killer: CivilizationInfo, tile: TileInfo) {
+        val assignedQuest = assignedQuests.filter {
+            it.assignee == killer.civName &&
+            it.questName == "Kill Camp" &&
+            it.data1.toFloat() == tile.position.x &&
+            it.data2.toFloat() == tile.position.y
+        }.firstOrNull()
+
+        if (assignedQuest != null) {
+            giveReward(assignedQuest)
+            assignedQuests.remove(assignedQuest)
+        }
+    }
+
+    /** Get notified that [killer] conquered [cityState] */
+    fun cityStateKillNotification(killer: CivilizationInfo, cityState: CivilizationInfo) {
+        val assignedQuest = assignedQuests.filter {
+            it.assignee == killer.civName &&
+                    it.questName == "Kill City State" &&
+                    it.data1 == cityState.civName
+        }.firstOrNull()
+
+        if (assignedQuest != null) {
+            giveReward(assignedQuest)
+            assignedQuests.remove(assignedQuest)
+        }
+    }
```

Comment:
This looks like the same function twice with minor differences, don't repeat yourself